### PR TITLE
feat(outgoingwebhookevent): adding api for query to fetch outgoing webhook events log

### PR DESCRIPTION
**Pull request by harsh-sharma-juspay**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
adding GET api for query to fetch outgoing webhook events log by specified filters
**Filter:**
```payment_id: String```
```event_id: Option<String>```
```refund_id: Option<String>```
```dispute_id: Option<String>```
```mandate_id: Option<String>```
```payment_method_id: Option<String>```
```attempt_id: Option<String>```

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

**Api Endpoint:** ```http://localhost:8080/analytics/v1/outgoing_webhook_event_logs```
curl attached below with sample response tested on local machine

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by harsh-sharma-juspay** [comment]
Curl:

```curl --location 'http://localhost:8080/analytics/v1/outgoing_webhook_event_logs?payment_id=pay_7Ipa9i5nyphFOqA5BGA1' \
--header 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:121.0) Gecko/20100101 Firefox/121.0' \
--header 'Accept: */*' \
--header 'Accept-Language: en-US,en;q=0.5' \
--header 'Accept-Encoding: gzip, deflate, br' \
--header 'Referer: https://app.hyperswitch.io/' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiZTE4ZjJlMjMtMWZlOC00OTIyLTllNGQtODI4MTNjYmNhMmJiIiwibWVyY2hhbnRfaWQiOiJtZXJjaGFudF8xNzAyNjQxMjUyIiwicm9sZV9pZCI6Im9yZ19hZG1pbiIsImV4cCI6MTcwNTA3MDY4NSwib3JnX2lkIjoib3JnXzZNTTNMcWFJdGJvVFhIdXhHaDFDIn0.gPoeiQinucoo7lEOG-u68lDM_H9zI_T4LCTQFWAVCSk' \
--header 'api-key: hyperswitch' \
--header 'Origin: https://app.hyperswitch.io' \
--header 'Connection: keep-alive' \
--header 'Sec-Fetch-Dest: empty' \
--header 'Sec-Fetch-Mode: cors' \
--header 'Sec-Fetch-Site: same-site' \
--header 'TE: trailers'
```

**Comment by harsh-sharma-juspay** [comment]
sample Response:
```
[
    {
        "merchant_id": "merchant_1702641252",
        "event_id": "pay_7Ipa9i5nyphFOqA5BGA1_payment_succeeded",
        "event_type": "payment_succeeded",
        "outgoing_webhook_event_type": "payment",
        "payment_id": "pay_7Ipa9i5nyphFOqA5BGA1",
        "refund_id": null,
        "attempt_id": null,
        "dispute_id": null,
        "payment_method_id": null,
        "mandate_id": null,
        "content": "{\"payment_id\":\"pay_7Ipa9i5nyphFOqA5BGA1\",\"merchant_id\":\"merchant_1702641252\",\"status\":\"succeeded\",\"amount\":6540,\"amount_capturable\":0,\"amount_received\":6540,\"connector\":\"stripe\",\"client_secret\":\"*** alloc::string::String ***\",\"created\":\"2024-01-05T07:30:54.728Z\",\"currency\":\"USD\",\"customer_id\":\"StripeCustomer\",\"description\":\"Its my first payment request\",\"refunds\":null,\"disputes\":null,\"mandate_id\":null,\"mandate_data\":null,\"setup_future_usage\":null,\"off_session\":null,\"capture_on\":null,\"capture_method\":\"automatic\",\"payment_method\":\"card\",\"payment_method_data\":{\"card\":{\"last4\":\"4242\",\"card_type\":null,\"card_network\":null,\"card_issuer\":null,\"card_issuing_country\":null,\"card_isin\":\"424242\",\"card_exp_month\":\"*** alloc::string::String ***\",\"card_exp_year\":\"*** alloc::string::String ***\",\"card_holder_name\":\"*** alloc::string::String ***\"}},\"payment_token\":null,\"shipping\":{\"address\":{\"city\":\"San Fransico\",\"country\":\"US\",\"line1\":\"*** alloc::string::String ***\",\"line2\":\"*** alloc::string::String ***\",\"line3\":\"*** alloc::string::String ***\",\"zip\":\"*** alloc::string::String ***\",\"state\":\"*** alloc::string::String ***\",\"first_name\":\"*** alloc::string::String ***\",\"last_name\":\"*** alloc::string::String ***\"},\"phone\":{\"number\":\"*** alloc::string::String ***\",\"country_code\":\"+91\"}},\"billing\":{\"address\":{\"city\":\"San Fransico\",\"country\":\"US\",\"line1\":\"*** alloc::string::String ***\",\"line2\":\"*** alloc::string::String ***\",\"line3\":\"*** alloc::string::String ***\",\"zip\":\"*** alloc::string::String ***\",\"state\":\"*** alloc::string::String ***\",\"first_name\":\"*** alloc::string::String ***\",\"last_name\":\"*** alloc::string::String ***\"},\"phone\":{\"number\":\"*** alloc::string::String ***\",\"country_code\":\"+91\"}},\"order_details\":null,\"email\":\"*****@example.com\",\"name\":\"*** alloc::string::String ***\",\"phone\":\"*** alloc::string::String ***\",\"return_url\":\"https://google.com/\",\"authentication_type\":\"no_three_ds\",\"statement_descriptor_name\":\"joseph\",\"statement_descriptor_suffix\":\"JS\",\"next_action\":null,\"cancellation_reason\":null,\"error_code\":null,\"error_message\":null,\"unified_code\":null,\"unified_message\":null,\"payment_experience\":null,\"payment_method_type\":\"credit\",\"connector_label\":null,\"business_country\":null,\"business_label\":\"default\",\"business_sub_label\":null,\"allowed_payment_method_types\":null,\"ephemeral_key\":{\"customer_id\":\"StripeCustomer\",\"created_at\":1704439854,\"expires\":1704443454,\"secret\":\"epk_58eb2fb1951d440e927f4740a7a00ce2\"},\"manual_retry_allowed\":false,\"connector_transaction_id\":\"pi_3OV7rnD5R7gDAGff0JmWCZNC\",\"frm_message\":null,\"metadata\":\"*** serde_json::value::Value ***\",\"connector_metadata\":null,\"feature_metadata\":null,\"reference_id\":\"pi_3OV7rnD5R7gDAGff0JmWCZNC\",\"payment_link\":null,\"profile_id\":\"pro_dJTN6uGsYIuKeeG9IsgV\",\"surcharge_details\":null,\"attempt_count\":1,\"merchant_decision\":null,\"merchant_connector_id\":\"mca_FmLwc8mWoZR22H1nt4Ce\",\"incremental_authorization_allowed\":null,\"authorization_count\":null,\"incremental_authorizations\":null}",
        "is_error": false,
        "error": null,
        "created_at_timestamp": "2024-01-05T07:30:57.638Z"
    }
]
```

File: crates/analytics/src/outgoing_webhook_event/core.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        ))
+        .into_report()
+        .attach_printable("SQL Analytics is not implemented for Outgoing Webhook Events"),
+        AnalyticsProvider::Clickhouse(pool) => {
+            get_outgoing_webhook_event(&merchant_id, req, pool).await
+        }
+        AnalyticsProvider::CombinedSqlx(_sqlx_pool, ckh_pool)
+        | AnalyticsProvider::CombinedCkh(_sqlx_pool, ckh_pool) => {
+            get_outgoing_webhook_event(&merchant_id, req, ckh_pool).await
+        }
```
Comment (lsampras) [inline crates/analytics/src/outgoing_webhook_event/core.rs:25]:
```suggestion
        AnalyticsProvider::Clickhouse(ckh_pool)
        | AnalyticsProvider::CombinedSqlx(_sqlx_pool, ckh_pool)
        | AnalyticsProvider::CombinedCkh(_sqlx_pool, ckh_pool) => {
            get_outgoing_webhook_event(&merchant_id, req, ckh_pool).await
        }
```

**Review by lsampras — COMMENTED** [review]
minor nit

**Review by ivor-juspay — DISMISSED** [review]

**Review by lsampras — DISMISSED** [review]

**Review by ShivanshMathurJuspay — APPROVED** [review]

**Review by sagarnaikjuspay — DISMISSED** [review]

**Review by sagarnaikjuspay — APPROVED** [review]

**Review by ivor-juspay — DISMISSED** [review]

**Review by lsampras — APPROVED** [review]

**Review by sagarnaikjuspay — APPROVED** [review]

**Pull request merged**